libkineto/src/ActivityProfiler.cpp

```diff
@@ -187,6 +191,40 @@ void ActivityProfiler::processCpuTrace(
 
   CpuGpuSpanPair& span_pair = recordTraceSpan(cpuTrace.span, cpuTrace.gpuOpCount);
   TraceSpan& cpu_span = span_pair.first;
+  // Update act.unique_op_id to correlate forward-op and backward-op.
```

Comment:
I suggest adding all this code into profiler_kineto.cpp (as part of #62553) and call it from finalizeCPUTrace.
Then, the only thing we need to add to GenericTraceActivity is a linkedFlowId or a pointer to the linked activity that can be returned by linkedActivity().
My overall plan for flows is to encode a flow ID and a flow type in a flow field (e.g. 4 bits for flow type). Profilers can register flow types and associate them with names to make it possible to link activities across separate profilers.
In this case you're only linking activities within a single profiler so it's a bit simpler.

Reply:
Got it. Made following change to "GenericTraceActivity.h":
Added "TraceActivity* linkedAct;" to point to its corresponding backward or forward op. Added "uint64_t linkId;" to give a pair of forward&backward ops same value, the high-4bits is link type.
And move correlating logic to "finalizeCPUTrace".